### ISSUE TITLE
Bug: [prefer-readonly] prefer-readonly should not consider accessors

### ISSUE THREAD

**Issue by Jamesernator** [typescript-eslint/typescript-eslint#8299]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.tsx&code=FAYwNghgzlAEBiB7RsDexawA4CcCWAbhAC4CmsEIIpMiOFVsAvLAMwDcwGD1t9AxJRDNYAFk4BfIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0Wls4CGAEwD2TeIXRRe0EdEjgwAXxBKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class Foo {
  private accessor acc = 3;

  accessor #acc = 4;
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/prefer-readonly": "error"
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Accessors should be ignored by `prefer-readonly`.

### Actual Result

A broken fix is done that adds `readonly` to the accessors:

```ts
class Foo {
  private accessor readonly acc = 3;

  accessor readonly #acc = 4;
}
```

### Additional Info

I will make a PR for this.
